packages/arb-bridge-peripherals/contracts/tokenbridge/libraries/BytesParser.sol

```diff
@@ -19,32 +19,51 @@
 pragma solidity ^0.6.11;
 
 import "arb-bridge-eth/contracts/libraries/BytesLib.sol";
-import "arb-bridge-eth/contracts/libraries/DebugPrint.sol";
 
-library BytesParserWithDefault {
+library BytesParser {
     using BytesLib for bytes;
 
-    function toUint8(bytes memory input, uint8 defaultValue) internal pure returns (uint8) {
+    function toUint8(bytes memory input) internal pure returns (bool success, uint8 res) {
         if (input.length == 0) {
-            return defaultValue;
+            success = false;
+            // return the default value of uint8
         } else {
             // TODO: try catch to handle error
-            return abi.decode(input, (uint8));
+            success = true;
+            res = abi.decode(input, (uint8));
         }
     }
 
-    function toString(bytes memory input, string memory defaultValue)
-        internal
-        pure
-        returns (string memory)
-    {
+    function toString(bytes memory input) internal pure returns (bool success, string memory res) {
         if (input.length == 0) {
-            return defaultValue;
+            success = false;
+            // return default value of string
         } else if (input.length == 32) {
-            return DebugPrint.bytes32string(input.toBytes32(0));
+            // TODO: can validate anything other than length and being null terminated?
+            if (input[31] != bytes1(0x00)) return (false, res);
+            else success = true;
+
+            // here we assume its a null terminated Bytes32 string
+            // https://github.com/ethereum/solidity/blob/5852972ec148bc041909400affc778dee66d384d/test/libsolidity/semanticTests/externalContracts/_stringutils/stringutils.sol#L89
+            // https://github.com/Arachnid/solidity-stringutils
+            uint256 len = 32;
+            while (input[len - 1] == bytes1(0x00)) {
```

Comment:
sanity checking my understanding here - we could never even hit the out bounds error here b/c the empty bytes array check is handled above?